File: 3_Palindrome/main.py
import sys
from antlr4 import *
from PalindromeLexer import PalindromeLexer
from PalindromeParser import PalindromeParser
 
def main(argv):
    # Reads from file
    if(len(argv) <= 1):
        print("Usare: 'python3 main.py test-file")
        return;
        
    input_stream = FileStream(argv[1])

    # Lexer step
    lexer = PalindromeLexer(input_stream)
    stream = CommonTokenStream(lexer)

    # Uncomment to print the tokens readed
    stream.fill()
    for token in stream.getTokens(0, input_stream.size):
        print(str(token))

    # Parser step
    parser = PalindromeParser(stream)
    tree = parser.parse_all()
    if(parser.getNumberOfSyntaxErrors() == 0): 
        print("SUCCESS: input string is parsed correctly")
        # Print the parsing tree
        print(tree.toStringTree(None, parser))
 
if __name__ == '__main__':
    main(sys.argv)

File: 3_Palindrome/main.js
var antlr4 = require('antlr4');
var PalindromeLexer = require('./PalindromeLexer').PalindromeLexer;
var PalindromeParser = require('./PalindromeParser').PalindromeParser;

var input = "aaabbbaaa"
var chars = new antlr4.InputStream(input);

// Lexer step
var lexer = new PalindromeLexer(chars);
var tokens  = new antlr4.CommonTokenStream(lexer);

// Uncomment to print the tokens readed
tokens.fill();
tokens.getTokens(0, input.length).forEach(token =>
    console.log(token.toString())
)

// Parser step
var parser = new PalindromeParser(tokens);
parser.buildParseTrees = true;
var tree = parser.parse_all();

// Print the parsing tree
if(tree.parser._syntaxErrors == 0){
    console.log("Parsing result = SUCCESS")
    console.log(tree.toStringTree(null, parser))
}
